src/components/screens/on-call-calendar-screen.tsx

```diff
@@ -70,22 +70,21 @@ const OnCallCalendarScreen = () => {
       const fetchedData = await onCallApi.listOnCallData({ year: year.toString() });
       setOnCallData(fetchedData);
     } catch (error) {
-      setError(`${strings.oncall.fetchFailed}, ${error}`);
+      setError(`${strings.oncall.fetchFailed}`);
+      setOnCallData([]);
     }
     setLoading(false);
   };
 
   /**
    * Finds the current on call employee and sets them in a state
    */
-  const getCurrentOnCallPerson = () => {
+  const getCurrentOnCallPerson = async () => {
+    const currentYear = DateTime.now().year;
     const currentWeek = DateTime.now().weekNumber;
-    const currentOnCallPerson = onCallData.find((item) => item.week === currentWeek)?.username;
-    if (currentOnCallPerson) {
-      setOnCallPerson(currentOnCallPerson);
-    } else {
-      setOnCallPerson(null);
-    }
+    const fetchedData = await onCallApi.listOnCallData({ year: currentYear.toString() });
```

Comment:
This needs error handling, try/catch.
Looks like previously it was just assuming the onCallData existed from the atom (I assume meaning the request is made on the home screen for the card). Whats the reason for this change?

I'm not aware of what bug this branch is fixing